### ISSUE TITLE
core dump found

### ISSUE THREAD

**Issue by HenryTangMain** [harrism/cuda_event_benchmark#1]
```
nvidia-smi -L
GPU 0: NVIDIA GeForce RTX 2070 SUPER (UUID: GPU-a9749610-a63a-2e57-256f-e06518ca613b)

Running ./cuda_event_bench
Run on (8 X 4000 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x4)
  L1 Instruction 32 KiB (x4)
  L2 Unified 256 KiB (x4)
  L3 Unified 8192 KiB (x1)
Load Average: 0.29, 0.48, 0.72
***WARNING*** CPU scaling is enabled, the benchmark real time measurements may be noisy and will incur extra overhead.
---------------------------------------------------------------------------------------------
Benchmark                                   Time             CPU   Iterations UserCounters...
---------------------------------------------------------------------------------------------
BM_EventCreate<true>                      453 us          453 us         1549 items_per_second=2.20914M/s
BM_EventCreate<false>                     318 us          318 us         2219 items_per_second=3.14594M/s
BM_EventPool<true>                        112 us          112 us         6169 items_per_second=8.94511M/s
BM_EventPool<false>                       112 us          112 us         6201 items_per_second=8.93344M/s
BM_EventRecord_MT<true>/threads:1        1098 us         1097 us          570 items_per_second=911.311k/s
BM_EventRecord_MT<true>/threads:2        1776 us         2996 us          238 items_per_second=333.81k/s
BM_EventRecord_MT<true>/threads:4        1784 us         4455 us          160 items_per_second=224.472k/s
BM_EventRecord_MT<true>/threads:8        1728 us         5641 us          120 items_per_second=177.276k/s
BM_EventRecord_MT<false>/threads:1        253 us          253 us         2759 items_per_second=3.95868M/s
BM_EventRecord_MT<false>/threads:2        482 us          961 us          724 items_per_second=1040.57k/s
BM_EventRecord_MT<false>/threads:4        508 us         1797 us          384 items_per_second=556.382k/s
BM_EventRecord_MT<false>/threads:8        632 us         4235 us          168 items_per_second=236.107k/s
BM_EventQuery<true>                       172 us          172 us         4163 items_per_second=5.81351M/s
BM_EventQuery<false>                      191 us          191 us         3813 items_per_second=5.22987M/s
BM_StreamWaitEvent<true>                  213 us          213 us         3200 items_per_second=4.69852M/s
BM_StreamWaitEvent<false>                 208 us          208 us         3416 items_per_second=4.80708M/s
Segmentation fault (core dumped)
```

**Comment by harrism** [comment]
@htang2012 what CUDA toolkit version?

**Comment by HenryTangMain** [comment]
Hi @harrism, here are the versions of cuda software stack:

cat /usr/local/cuda/version.json
{
   "cuda" : {
      "name" : "CUDA SDK",
      "version" : "11.3.0"
   },
   "cuda_cudart" : {
      "name" : "CUDA Runtime (cudart)",
      "version" : "11.3.58"
   },
   "cuda_cuobjdump" : {
      "name" : "cuobjdump",
      "version" : "11.3.58"
   },
   "cuda_cupti" : {
      "name" : "CUPTI",
      "version" : "11.3.58"
   },
   "cuda_cuxxfilt" : {
      "name" : "CUDA cu++ filt",
      "version" : "11.3.58"
   },
   "cuda_demo_suite" : {
      "name" : "CUDA Demo Suite",
      "version" : "11.3.58"
   },
   "cuda_gdb" : {
      "name" : "CUDA GDB",
      "version" : "11.3.58"
   },
   "cuda_memcheck" : {
      "name" : "CUDA Memcheck",
      "version" : "11.3.58"
   },
   "cuda_nsight" : {
      "name" : "Nsight Eclipse Plugins",
      "version" : "11.3.58"
   },
   "cuda_nvcc" : {
      "name" : "CUDA NVCC",
      "version" : "11.3.58"
   },
   "cuda_nvdisasm" : {
      "name" : "CUDA nvdisasm",
      "version" : "11.3.58"
   },
   "cuda_nvml_dev" : {
      "name" : "CUDA NVML Headers",
      "version" : "11.3.58"
   },
   "cuda_nvprof" : {
      "name" : "CUDA nvprof",
      "version" : "11.3.58"
   },
   "cuda_nvprune" : {
      "name" : "CUDA nvprune",
      "version" : "11.3.58"
   },
   "cuda_nvrtc" : {
      "name" : "CUDA NVRTC",
      "version" : "11.3.58"
   },
   "cuda_nvtx" : {
      "name" : "CUDA NVTX",
      "version" : "11.3.58"
   },
   "cuda_nvvp" : {
      "name" : "CUDA NVVP",
      "version" : "11.3.58"
   },
   "cuda_samples" : {
      "name" : "CUDA Samples",
      "version" : "11.3.58"
   },
   "cuda_sanitizer_api" : {
      "name" : "CUDA Compute Sanitizer API",
      "version" : "11.3.58"
   },
   "fabricmanager" : {
      "name" : "Fabric Manager",
      "version" : "465.19.01"
   },
   "libcublas" : {
      "name" : "CUDA cuBLAS",
      "version" : "11.4.2.10064"
   },
   "libcufft" : {
      "name" : "CUDA cuFFT",
      "version" : "10.4.2.58"
   },
   "libcurand" : {
      "name" : "CUDA cuRAND",
      "version" : "10.2.4.58"
   },
   "libcusolver" : {
      "name" : "CUDA cuSOLVER",
      "version" : "11.1.1.58"
   },
   "libcusparse" : {
      "name" : "CUDA cuSPARSE",
      "version" : "11.5.0.58"
   },
   "libnpp" : {
      "name" : "CUDA NPP",
      "version" : "11.3.3.44"
   },
   "libnvjpeg" : {
      "name" : "CUDA nvJPEG",
      "version" : "11.4.1.58"
   },
   "nsight_compute" : {
      "name" : "Nsight Compute",
      "version" : "2021.1.0.18"
   },
   "nsight_systems" : {
      "name" : "Nsight Systems",
      "version" : "2021.1.3.14"
   },
   "nvidia_driver" : {
      "name" : "NVIDIA Linux Driver",
      "version" : "465.19.01"
   }
}

**Comment by harrism** [comment]
I have a local repro. Debugging.

**Comment by harrism** [comment]
@htang2012 can you try out the fix in #2? If it works for you I will merge.
